### PR TITLE
fix `copy mailing address` bugs

### PR DESCRIPTION
**Pull request by erikphansen**
## Description
Adds proper cleanup up address data when copying a mailing address to use it as a residential address and when comparing the two types of addresses to see if they are the same.

## Testing done
confirmed locally that:
- clicking the checkbox uses the mailing address as the residential address and attempting to save the new address does not result in form validation errors
- the checkbox is prechecked when you open the residential address edit modal when the residential address is already the same as the mailing address

## Screenshots
**Copying and saving works**
copy-and-save

**The checkbox is preselected when it should be**
preselected

## Acceptance criteria
- [ ]

## Definition of done
- [ ] Events are logged appropriately
- [ ] Documentation has been updated, if applicable
- [x] A link has been provided to the originating GitHub issue (or connected to it via ZenHub)
- [x] No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs

### CODE REVIEW

File: src/platform/user/profile/vet360/components/AddressField/AddressEditModal.jsx [hunk 1]
```diff
@@ -113,7 +113,7 @@ class AddressEditModal extends React.Component {
     const newAddressValue = { ...this.props.field.value, ...mailingAddress };
     if (useNewAddressForm) {
       this.props.onChangeFormDataAndSchemas(
-        newAddressValue,
+        this.transformInitialFormValues(newAddressValue),
```
Comment (erikphansen) [inline src/platform/user/profile/vet360/components/AddressField/AddressEditModal.jsx]: this cleans up the mailing address (removes null values) before using it as the residential address.

File: src/platform/user/profile/vet360/containers/CopyMailingAddress.jsx [hunk 2]
```diff
@@ -68,7 +69,7 @@ export function mapStateToProps(state, ownProps) {
     }
     if (useNewAddressForm) {
       return isEqual(
-        pick(mailingAddress, ADDRESS_PROPS),
+        pickBy(pick(mailingAddress, ADDRESS_PROPS), e => !!e),
```
Comment (erikphansen) [inline src/platform/user/profile/vet360/containers/CopyMailingAddress.jsx]: this removes undefined/null values from the mailing address before comparing it to the residential address.

**Review by bradl3yC — APPROVED** [review]

**Review by ncksllvn — APPROVED** [review]

File: src/platform/user/profile/vet360/containers/CopyMailingAddress.jsx (hunk 2 above)
Comment (bkjohnson) [reply src/platform/user/profile/vet360/containers/CopyMailingAddress.jsx]: Can you add a unit test for this?

Comment (erikphansen) [reply src/platform/user/profile/vet360/containers/CopyMailingAddress.jsx]: Done. Thanks for keeping me honest :)

**Review by bkjohnson — APPROVED** [review]
:+1:

**Pull request merged**